app/Http/Controllers/ApiPagesController.php

```diff
@@ -41,4 +41,45 @@ public function createNewApp()
     {
         return view('api.pages.mynewapp');
     }
+
+    /**
+     * Generate a token for user.
+     *
+     * @param int $userId
+     *
+     * @return string
+     */
+    public function generateToken()
+    {
+        $appSecret = getenv('APP_SECRET');
+        $jwtAlgorithm = getenv('JWT_ALGORITHM');
+        $timeIssued = time();
+        $serverName = getenv('SERVERNAME');
+        $tokenId = base64_encode(getenv('TOKENID'));
+        $token = [
+            'iss'  => $serverName,   //Issuer: the server name
+            'iat'  => $timeIssued,   // Issued at: time when the token was generated
+            'jti'  => $tokenId,          // Json Token Id: an unique identifier for the token
+            'nbf'  => $timeIssued, //Not before time
+            'exp'  => $timeIssued + 60 * 60 * 24 * 30, // expires in 30 days
+        ];
+        
+        return JWT::encode($token, $appSecret, $jwtAlgorithm);
+    }
+
+    /**
+     * This method post the details from the user into the database
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function postAppDetails(Request $request)
```

Comment:
**Code Quality - Critical**
@andela-araimi Good method name, but the method is confusing and incomplete, the annotation says it returns `\Illuminate\Http\Response` but there is nothing being returned in this method. This method is meant to redirect to a new page with a success message or redirect back to the same page with an error message. And also there is no validation for this form post. Please fix all of this.